web/components/editor.tsx

```diff
@@ -56,6 +63,71 @@ export function useTextEditor(props: {
           class: clsx('no-underline !text-indigo-700', linkClass),
         },
       }),
+      Mention.configure({
+        HTMLAttributes: {
+          class: clsx('not-prose text-indigo-700', linkClass),
+        },
+        // TODO: do a Next link instead of raw <a>
+        renderLabel: ({ options, node }) =>
+          [
+            'a',
+            { href: node.attrs.label },
+            `${options.suggestion.char}${node.attrs.label ?? node.attrs.id}`,
+          ] as any,
+        suggestion: {
+          items: ({ query }) =>
+            users
+              .filter((u) => searchInAny(query, u.username, u.name))
+              .slice(0, 5),
+          render: () => {
+            let component: any
+            let popup: any
+            return {
+              onStart: (props) => {
+                component = new ReactRenderer(MentionList, {
+                  props,
+                  editor: props.editor,
+                })
+                if (!props.clientRect) {
+                  return
+                }
+
+                popup = tippy('body', {
+                  getReferenceClientRect: props.clientRect as any,
+                  appendTo: () => document.body,
+                  content: component.element,
+                  showOnCreate: true,
+                  interactive: true,
+                  trigger: 'manual',
+                  placement: 'bottom-start',
+                })
+              },
+              onUpdate(props) {
+                component.updateProps(props)
+
+                if (!props.clientRect) {
+                  return
+                }
+
+                popup[0].setProps({
```

Comment:
It's a little mysterious to use popup[0]. Maybe save a reference directly to popup[0]?